src/global/icon-cache.ts

```diff
@@ -1,53 +1,51 @@
+const CACHE_NAME = '@limetech/lime-elements/icons';
+
 export class IconCache {
     /*
      * Cache of all loaded SVGs
      */
-    private cache = {};
+    private cache: Promise<Cache>;
 
-    /*
-     * Contains resolve functions for all unresolved promises that are waiting for SVG data.
-     */
-    private resolveFunctions = {};
+    private promises: Record<string, Promise<void>> = {};
+
+    constructor() {
+        this.cache = caches.open(CACHE_NAME);
+    }
 
     /**
      * Get icon data from the cache
      * @param {string} name name of the icon
      * @param {string} path path on the server where the assets are located
      * @returns {string} svg markup
      */
-    public async get(name, path = '') {
-        if (!this.cache[name]) {
-            this.cache[name] = await this.getIcon(name, path);
+    public async get(name: string, path: string = ''): Promise<string> {
+        const cache = await this.cache;
+        const url = this.getUrl(name, path);
+
+        let response = await cache.match(url);
+        if (!response) {
+            response = await this.fetchData(url, cache);
         }
 
-        return this.cache[name];
+        return this.getIcon(response);
     }
 
-    /*
-     * Creates and returns a promise that will be resolved when SVG data is available
-     */
-    private getIcon(name, path) {
-        return new Promise((resolve) => {
-            if (!this.resolveFunctions[name]) {
-                this.resolveFunctions[name] = [];
-                this.fetchData(name, path);
-            }
-
-            this.resolveFunctions[name].push(resolve);
-        });
+    private async fetchData(url: string, cache: Cache): Promise<Response> {
+        let requestPromise = this.promises[url];
```

Comment:
I'm probably missing something, but I'm not sure why we need this.promises.

Couldn't we do away with the condition here as well as `this.promises` entirely?

Reply:
I see, that makes sense.

Let's ship it then 🚢